clap_complete/src/dynamic/completer.rs

```diff
@@ -449,6 +462,34 @@ fn complete_path(
     completions
 }
 
+fn complete_custom_valuehint(value: Result<String, OsString>, arg: &clap::Arg) -> Vec<OsString> {
+    let values = Arc::new(Mutex::new(Vec::new()));
+    let values_clone = Arc::clone(&values);
+    let arg_clone = arg.clone();
+    let value_clone = value.clone();
+
+    let custom = thread::spawn(move || {
```

Comment:
Why are we spawning a thread?

Reply:
Just to prevent the current thread from panic. Because there are other possible completions that will be generated.